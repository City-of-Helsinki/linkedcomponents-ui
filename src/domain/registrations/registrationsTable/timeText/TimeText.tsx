import React from 'react';
import { useTranslation } from 'react-i18next';

import useLocale from '../../../../hooks/useLocale';
import useTimeFormat from '../../../../hooks/useTimeFormat';
import formatDate from '../../../../utils/formatDate';
import styles from './timeText.module.scss';

export type TimeTextProps = {
  startTime: Date | null | undefined;
  endTime: Date | null | undefined;
};

const TimeText: React.FC<TimeTextProps> = ({ startTime, endTime }) => {
  const timeFormat = useTimeFormat();
  const locale = useLocale();
  const { t } = useTranslation();

  if (!startTime && !endTime) {
    return '-';
  }

  const startText =
    startTime &&
    t('eventsPage.datetime', {
      date: formatDate(startTime),
      time: formatDate(startTime, timeFormat, locale),
    });
  const endText =
    endTime &&
    t('eventsPage.datetime', {
      date: formatDate(endTime),
      time: formatDate(endTime, timeFormat, locale),
    });

  return (
    <div aria-label={[startText, endText].join(' – ').trim()}>
      <div className={styles.timeText}>
        {startTime && <span>{startText}</span>}
        <span> – </span>
      </div>
      {endTime && <div className={styles.timeText}>{endText}</div>}
    </div>
  );
};

export default TimeText;
