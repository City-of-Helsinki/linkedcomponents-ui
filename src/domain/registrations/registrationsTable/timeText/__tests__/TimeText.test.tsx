import { render } from '@testing-library/react';
import React from 'react';

import TimeText, { TimeTextProps } from '../TimeText';

const renderComponent = (props: TimeTextProps) =>
  render(<TimeText {...props} />);

test('should render correct text when startTime and endTime are defined', () => {
  const { container } = renderComponent({
    startTime: new Date('2021-07-13T05:00:00.000000Z'),
    endTime: new Date('2021-08-15T12:15:00.000000Z'),
  });
  expect(container).toMatchSnapshot();
});

test('should render correct text when only startTime is defined', () => {
  const { container } = renderComponent({
    startTime: new Date('2021-07-13T05:00:00.000000Z'),
    endTime: null,
  });
  expect(container).toMatchSnapshot();
});

test('should render correct text when only endTime is defined', () => {
  const { container } = renderComponent({
    endTime: new Date('2021-08-15T12:15:00.000000Z'),
    startTime: null,
  });
  expect(container).toMatchSnapshot();
});

test('should render correct text when startTime and endTime are not defined', () => {
  const { container } = renderComponent({
    endTime: null,
    startTime: null,
  });
  expect(container).toMatchSnapshot();
});
