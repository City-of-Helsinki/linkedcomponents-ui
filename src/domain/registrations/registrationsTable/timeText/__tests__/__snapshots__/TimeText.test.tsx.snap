// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render correct text when only endTime is defined 1`] = `
<div>
  <div
    aria-label="– 15.8.2021 klo 12.15"
  >
    <div
      class="timeText"
    >
      <span>
         – 
      </span>
    </div>
    <div
      class="timeText"
    >
      15.8.2021 klo 12.15
    </div>
  </div>
</div>
`;

exports[`should render correct text when only startTime is defined 1`] = `
<div>
  <div
    aria-label="13.7.2021 klo 05.00 –"
  >
    <div
      class="timeText"
    >
      <span>
        13.7.2021 klo 05.00
      </span>
      <span>
         – 
      </span>
    </div>
  </div>
</div>
`;

exports[`should render correct text when startTime and endTime are defined 1`] = `
<div>
  <div
    aria-label="13.7.2021 klo 05.00 – 15.8.2021 klo 12.15"
  >
    <div
      class="timeText"
    >
      <span>
        13.7.2021 klo 05.00
      </span>
      <span>
         – 
      </span>
    </div>
    <div
      class="timeText"
    >
      15.8.2021 klo 12.15
    </div>
  </div>
</div>
`;

exports[`should render correct text when startTime and endTime are not defined 1`] = `
<div>
  -
</div>
`;
